src/namespace/privilegednamespace.js

```diff
@@ -287,4 +362,165 @@ export class PrivilegedNamespace extends CustomNamespace {
+        if (!TypeCheck.isFunction(callback)) {
+            logger.debug(`callback is not a function.`);
+            return;
+        }
+
+        let {uuid, debateId} = banObj;
+        if (!TypeCheck.isString(uuid) || !TypeCheck.isInteger(debateId)) {
+            logger.debug('Invalid arguments for banUser');
+            callback(false);
+            return;
+        }
+
+        const debate = this.getActiveDebate(debateId);
+        if (debate == null) {
+            logger.warn(`Debate with id (${debateId}) not found. Can't ban properly.`);
+            callback(false);
+            return;
+        }
+
+        // Ban the device in the database
+        let res = await dbManager.banDevice(uuid, socket.username);
+        if (res === false) { // A ban should always work.. We add an uuid to the database if not found.
+            logger.error(`Cannot ban device with uuid ${uuid}`);
+            callback(false);
+            return;
+        }
+
+        // Disconnect the client
+        let client = debate.getClient(uuid);
+        if (client == null) {
+            logger.debug(`Client with uuid (${uuid}) is not connected`)
+        } else {
+            // Inform the client he is banned
+            client.socket.emit('banned', ErrorMessage.BAN_MESSAGE);
+            client.socket.disconnect();
+        }
+
+        logger.info(`User (${socket.username}) banned the device with uuid ${uuid}`);
+        callback(true);
+    };
+
+    /**
+     * Unban a user from all admin future debates
+     * unbanObj contains the required information (uuid)
+     */
+    unbanUser = (socket) => async (unbanObj, callback) => {
+        logger.debug(`unbanUser received from user (${socket.username}), id(${socket.id})`);
+
+        if (!TypeCheck.isFunction(callback)) {
+            logger.debug(`callback is not a function.`);
+            return;
+        }
+
+        let {uuid} = unbanObj;
+        if (!TypeCheck.isString(uuid)) {
+            logger.debug('Invalid arguments for unbanUser');
+            callback(false);
+            return;
+        }
+
+        // Check if the device is banned by this user
+        let isBanned = await dbManager.isDeviceBanned(uuid, socket.username);
+        if (!isBanned) {
+            logger.info(`Device with uuid (${uuid}) is not banned`);
+            callback(true);
+            return;
+        }
+
+        // Ban the device in the database
+        let res = await dbManager.unbanDevice(uuid, socket.username);
+        if (res === false) { // We know the device exists and was banned by this specific user...
+            logger.error(`Couldn't unban device with uuid ${uuid}`);
+            callback(false);
+            return;
+        }
+
+        logger.info(`User (${socket.username}) unbanned the device with uuid ${uuid}`);
+        callback(true);
+    };
+
+    /**
+     * Approve a suggestion with the specified id and debate
+     * approveObj contains the required information (debateId and suggestionId)
```

Comment:
At line 465 you run `let {suggestionId, debateId} = approveObj`. `suggestionId`and `debateId` are not in th same order as in your function doc. If there is an important order for `approveObj` you must specify it in function doc.

Reply:
Order doesn't matter as it is an object. I still updated it for readability in 2cab5a0.